INT-3665: Remove deprecations and resolve issues

artembilan opened the pull request:
https://jira.spring.io/browse/INT-3665

artembilan committed:
INT-3665: Remove deprecations and resolve issues

https://jira.spring.io/browse/INT-3665

artembilan committed:
Introduce `...ExpressionString(String)` setter

Some further fixes and polishing

garyrussell commented on spring-integration-amqp/src/main/java/org/springframework/integration/amqp/outbound/AmqpOutboundEndpoint.java:
```diff
@@ -111,23 +104,17 @@ public void setExchangeName(String exchangeName) {
-	public void setExchangeNameExpression(String exchangeNameExpression) {
-		Assert.hasText(exchangeNameExpression);
-		this.exchangeNameExpression = expressionParser.parseExpression(exchangeNameExpression);
+	public void setExchangeNameExpression(Expression exchangeNameExpression) {
```
I am not so sure this is a good idea any more; people are more and more using Java rather than XML for configuration. We recently added String support for expressions in the DSL; it seems onerous to me to force users to parse expression strings.

artembilan replied:
Actually we need such an `Expression` setter exactly for the DSL, just because we have there `ValueExpression` and `FunctionExpression`. 
For the `String` case I'd better introduce something like `setExchangeNameExpressionString(String exchangeNameExpression)`.
WDYT?

garyrussell replied:
Yes, that's fine; although I think in some cases you used `setExpressionExchangeName()`, or similar (when adding the expression version).

artembilan replied:
Not sure what you mean regarding "normalize all these things", because that is exactly what I'm doing here with this de-deprecation and setter argument type changing.
The `setExpressionExchangeName()` has been introduced for the parser limitation with the Java Bean nature. That's why we are renaming them now to the more readable name, which was with the existing setter for `String`.

Having all that confuse, I suggest to introduce a new `...ExpressionString(String)` setter do not confuse parser and allow to Java-config users have some convenient during their routine.

With your `socketExpression` introduction I see that you agreed here with me and I can go ahead with the proposed fix not waiting for the 5.0. 4.3 should be good enough for such a refactoring change.
WDYT?

garyrussell replied:
Yes; that's what I meant (like in the UDP PR).

My "normalize" comment means to standardize on that pattern everywhere.

I guess we can do it with 4.3; deprecate the non-conformant methods and remove them in 5.0.

artembilan replied:
M-m-m. That is exactly what I'm trying to do here:
* make `...Expression()` setter based on `Expression`
* deprecate `setExpression...()` setter.
* and after this our discussion introduce a new `...ExpressionString()` setter.

Am I missing anything else?

garyrussell replied:
Right - you are not missing anything; I am talking about the larger task - such as fixing the `RedisOutboundGateway`

`setCommandExpression` Vs. `setExpressionCommand` Vs. `setCommandExpressionString`.

artembilan committed:
Fixes according Travis report

Conflicts:
	spring-integration-core/src/main/java/org/springframework/integration/config/xml/AbstractConsumerEndpointParser.java

artembilan commented:
Pushed

garyrussell commented on spring-integration-amqp/src/main/java/org/springframework/integration/amqp/outbound/AmqpOutboundEndpoint.java:
```diff
@@ -111,23 +104,26 @@ public void setExchangeName(String exchangeName) {
 		this.exchangeName = exchangeName;
 	}
 
+	public void setExchangeNameExpression(Expression exchangeNameExpression) {
```
javadoc + `@since`

garyrussell replied:
`Assert.notNull()`

artembilan replied:
This method has been here for a while. Right now it is changed only with the expected type.
Although I won't mind to add JavaDocs...
The `notNull()` isn't necessary. See `doInit()` in this class.

garyrussell replied:
A new parameter type == new method, so `@since 4.3` is correct.

garyrussell commented on spring-integration-amqp/src/main/java/org/springframework/integration/amqp/outbound/AmqpOutboundEndpoint.java:
```diff
@@ -111,23 +104,26 @@ public void setExchangeName(String exchangeName) {
 
 	/**
-	 * Temporary, will be changed to {@link #setExchangeNameExpression} in a future release.
+	 * @deprecated in favor of {@link #setExchangeNameExpressionString(String)}.
```
in favor of `setExchangeNameExpression`

garyrussell commented on spring-integration-core/src/main/java/org/springframework/integration/endpoint/ExpressionMessageProducerSupport.java:
```diff
@@ -31,29 +30,30 @@
 
 	private volatile EvaluationContext evaluationContext;
 
+	public void setPayloadExpression(Expression payloadExpression) {
```
javadoc + `@since`

garyrussell commented on spring-integration-core/src/main/java/org/springframework/integration/endpoint/ExpressionMessageProducerSupport.java:
```diff
@@ -31,29 +30,30 @@
 
 	/**
-	 * Temporary, will be changed to {@link #setPayloadExpression} in a future release.
+	 * @deprecated in favor of {@link #setPayloadExpressionString(String)}.
```
`setPayloadExpression`

garyrussell commented on spring-integration-file/src/main/java/org/springframework/integration/file/remote/gateway/AbstractRemoteFileOutboundGateway.java:
```diff
@@ -373,24 +373,26 @@ public void setMputFilter(FileListFilter<File> filter) {
 		this.mputFilter = filter;
 	}
 
+	public void setRenameExpression(Expression renameExpression) {
```
ditto here and below

garyrussell commented on spring-integration-file/src/test/java/org/springframework/integration/file/remote/gateway/RemoteFileOutboundGatewayTests.java:
```diff
@@ -1009,214 +1011,216 @@ public Object answer(InvocationOnMock invocation) throws Throwable {
-		super(sessionFactory, Command.toCommand(command), expression);
-		this.setBeanFactory(mock(BeanFactory.class));
-	}
+	class TestRemoteFileOutboundGateway extends AbstractRemoteFileOutboundGateway<TestLsEntry> {
```
I am not sure why you made these inner classes, but in any case, this one should be `static`.

garyrussell commented on spring-integration-file/src/test/java/org/springframework/integration/file/remote/gateway/RemoteFileOutboundGatewayTests.java:
```diff
@@ -1009,214 +1011,216 @@ public Object answer(InvocationOnMock invocation) throws Throwable {
-	public boolean isDirectory() {
-		return this.dir;
-	}
+	class TestLsEntry extends AbstractFileInfo<TestLsEntry> {
```
and this; and the next

garyrussell commented on spring-integration-gemfire/src/main/java/org/springframework/integration/gemfire/store/GemfireMessageStore.java:
```diff
@@ -29,9 +32,6 @@
 
-import com.gemstone.gemfire.cache.Cache;
-import com.gemstone.gemfire.cache.Region;
-
```
:smile:

artembilan replied:
Yep! Just has fixed import settings on Windows. Made them the same like on Mac:
![imports](https://cloud.githubusercontent.com/assets/1071590/11669202/77911200-9dc9-11e5-8a32-254a122320bf.png)

garyrussell commented on spring-integration-redis/src/main/java/org/springframework/integration/redis/outbound/RedisOutboundGateway.java:
```diff
@@ -71,23 +71,26 @@ public void setArgumentsSerializer(RedisSerializer<?> serializer) {
 		this.argumentsSerializer = (RedisSerializer<Object>) serializer;
 	}
 
+	public void setCommandExpression(Expression commandExpression) {
```
javadoc etc (and `setCommandExpression` below)

garyrussell commented:
That's all - mostly minor polish, javadocs.

artembilan committed:
Address PR comments

artembilan commented:
Pushed.

garyrussell commented:
Merged as cf528c0b5dfb21e9ea6383d54c2541ca87281820
